source/resolve-conflicts.ts

```diff
@@ -0,0 +1,129 @@
+import {Editor, Doc} from 'codemirror';
+
+interface CodeMirrorInstance extends Editor, Doc {
+}
+
+class Resolver {
+	editor: CodeMirrorInstance
+
+	conflictNumber = 1
+
+	constructor() {
+		const codeMirrorElement = document.querySelector('.CodeMirror');
+		this.editor = (codeMirrorElement as any).CodeMirror;
+
+		// Listen to swapDoc event in order to find out file change
+		this.editor.on('swapDoc', () => {
+			// Add 200 milisecond delay, we need keep calm until codemirror updates its content
+			setTimeout(() => {
+				const conflicts = document.querySelector('.CodeMirror .rgh-conflict');
+				if (conflicts === null) {
+					this.processConflicts();
+				}
+			}, 200);
+		});
+	}
+
+	// Process editor lines and find conflict lines, assign class to each of lines
+	// and add resolver widget to the first line of conflict
+	processConflicts() {
+		const lines = document.querySelectorAll('.CodeMirror .conflict-gutter-marker');
+		let stepBranch = '';
+		for (const line of lines) {
+			const lineNumber = Number(line.parentElement.parentElement.textContent) - 1;
+			let cl = `rgh-conflict c-${this.conflictNumber} `;
+			if (line.classList.contains('js-start')) {
+				cl += 'rgh-separator rgh-start rgh-incoming-branch rgh-current-branch';
+				stepBranch = 'rgh-incoming-branch';
+				this.addWidgetToLine(this.conflictNumber, lineNumber);
+			} else if (line.classList.contains('js-middle')) {
+				cl += 'rgh-separator rgh-middle rgh-incoming-branch rgh-current-branch';
+				stepBranch = 'rgh-current-branch';
+			} else if (line.classList.contains('js-end')) {
+				cl += 'rgh-separator rgh-end rgh-incoming-branch rgh-current-branch';
+				this.conflictNumber += 1;
+			} else if (line.classList.contains('js-line')) {
+				cl += 'rgh-middle ' + stepBranch;
+			}
+
+			this.editor.addLineClass(lineNumber, '', cl);
+		}
+	}
```

Comment:
Do you think this step can be skipped? Instead of adding lines here, perhaps this step should only add the widget. Then `acceptBranch` would look at the following lines and apply the requested operation on the fly.

Perhaps `conflictNumber` can be skipped too since `acceptBranch` would just look at the following lines instead of querying the dom by class.